python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -447,10 +447,16 @@ def get_best_checkpoint(
         mode = self._validate_mode(mode)
 
         checkpoint_paths = self.get_trial_checkpoints_paths(trial, metric)
+
         if not checkpoint_paths:
             logger.error(f"No checkpoints have been found for trial {trial}.")
             return None
 
+        # Filter out nan. Sorting nan values leads to undefined behavior.
+        checkpoint_paths = [
+            (path, metric) for path, metric in checkpoint_paths if not is_nan(metric)
+        ]
+
```

Comment:
I think we should also check whether the filtered list is empty.

In addition, we can also have a fallback option when all metric values are `nan`. For example, if all are `nan`, return the latest checkpoint.

Reply:
I think we could just move the filtering above the `if not checkpoint_paths` check.

Reply:
Updated!